libfwupdplugin/fu-device.c

```diff
@@ -2155,6 +2157,54 @@ fu_device_fixup_vendor_name(FuDevice *self)
 	}
 }
 
+static gchar *
+fu_device_fixup_vendor(const gchar *vendor)
+{
+	g_autoptr(GString) str = NULL;
+	const gchar *suffixes[] = {"Ltd.",
+				   "Ltd",
+				   "Limited",
+				   "Inc.",
+				   "Inc",
+				   "Co.",
+				   "Co",
+				   "Corporation",
+				   "Corp",
+				   "corp.",
+				   "Incorporated",
+				   "GmbH",
+				   NULL};
+	struct {
+		const gchar *old;
+		const gchar *new;
+	} vendor_names[] = {{"ADATA Technology Co., Ltd.", "ADATA"},
+			    {"Chicony Electronics Co.,Ltd.", "Chicony"},
+			    {"LENOVO", "Lenovo"},
+			    {"Micron Technology Inc", "Micron"},
+			    {"Samsung Electronics Co Ltd", "Samsung"},
+			    {"VIA Labs, Inc.", "VIA"},
+			    {NULL, NULL}};
```

Comment:
Which vendor prompted this?  I wonder if some vendors might not be comfortable with their name being modified when it's displayed.

Reply:
The root problem is that vendors are using the legal name in the NVMe or OUI and using the "friendly name" in the product string. So we get something like `Samsung Electronics Co Ltd SAMSUNG FOOBARBAZ` which is just terrible.

Reply:
Should this fixup only be for NVME/ATA then perhaps?

Reply:
The other two is Chicony, which is the pixart plugin, and Lenovo which is the uefi_capsule plugin.

Reply:
Hmm. I still don't think it's the right precedent to do for all plugins.
A few local fixups to plugins are better in my opinion.
 I don't want legal troubles in fwupd because it shows Foo instead of the legal name Foo Limited.

Reply:
How about we just do the `"Samsung Electronics Co Ltd" -> "Samsung"` one in the NVME plugin as it's the worst offender in the `Samsung Electronics Co Ltd SAMSUNG FOOBARBAZ` case.

Reply:
sounds good